lib/awful/screenshot.lua

```diff
@@ -0,0 +1,884 @@
+      return nil
+    end
+
+  else
+    -- No directory argument means use the default. Technically an outrageously
+    -- invalid argument (i.e. not even a string) currently falls back to the
+    -- default as well.
+    return get_default_dir()
+  end
+
+end
+
+-- Internal function to sanitize a prefix string
+--
+-- Currently only strips all path separators ('/'). Allows for empty prefix.
+local function check_prefix(prefix)
+  -- Maybe add more sanitizing eventually
+  if prefix and type(prefix) == "string" then
+    prefix = string.gsub(prefix, '/', '')
+    return prefix
+  end
+  return get_default_prefix()
+end
+
+-- Internal routine to verify that a filepath is valid.
+local function check_filepath(filepath)
+
+  -- This module is forcing png for now. In the event of adding more
+  -- options, this function is basically unchanged, except for trying
+  -- to match a regex for each supported format (e.g. (png|jpe?g|gif|bmp))
+  -- NOTE: we should maybe make this case insensitive too?
+  local fname_start, fname_end = string.find(filepath,'/[^/]+%.png$')
+
+  if fname_start and fname_end then
+    local directory = string.sub(filepath, 1, fname_start)
+    local file_name = string.sub(filepath, fname_start + 1, fname_end)
+    directory = check_directory(directory)
+    if directory then
+      return directory .. file_name
+    else
+      return nil
+    end
+  end
+
+  return nil
+
+end
+
+-- Internal function to attempt to parse a filepath into directory and prefix.
+-- Returns directory, prefix if both, directory if no prefix, or nil if invalid
+local function parse_filepath(filepath)
+
+  -- Same remark as above about adding more image formats in the future
+  local fname_start, fname_end = string.find(filepath,'/[^/]+%.png$')
+
+  if fname_start and fname_end then
+
+    local directory = string.sub(filepath, 1, fname_start)
+    local file_name = string.sub(filepath, fname_start + 1, fname_end)
+
+    if fname_end - fname_start > 14 + 4 then
+
+      local base_name = string.sub(file_name, 1, #file_name - 4)
+      -- Is there a repeat count in Lua patterns, like {14} for most regexes?
+      local date_start, date_end = string.find(base_name, '%d%d%d%d%d%d%d%d%d%d%d%d%d%d$')
+
+      if date_start and date_end then
+        return directory, string.sub(base_name, 1, date_start - 1)
+      end
+
+    end
+
+    return directory
+
+  end
+
+  return nil
+
+end
+
+-- Internal function to configure the directory/prefix pair for any call to the
+-- screenshot API.
+--
+-- This supports using a different directory or filename prefix for a particular
+-- use by passing the optional arguments. In the case of an initalized object
+-- with  no arguments passed, the stored configuration parameters are quickly
+-- returned. This also technically allows the user to take a screenshot in an
+-- unitialized state and without passing any arguments.
+local function configure_path(directory, prefix)
+
+  local d, p
```

Comment:
i was talking of these specifically

Reply:
OK, I'll make them `dir, prfx` or something.